src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py

```diff
@@ -344,7 +352,7 @@ def __call__(
                 noise_pred = noise_pred_uncond + guidance_scale * (noise_pred_text - noise_pred_uncond)
 
             # compute the previous noisy sample x_t -> x_t-1
-            latents = self.scheduler.step(noise_pred, t, latents, **extra_step_kwargs).prev_sample
+            latents = self.scheduler.step(noise_pred, t, latents, generator=generator, **extra_step_kwargs).prev_sample
```

Comment:
all schedulers now accept `generator` in `step`

Reply:
For this PR could we do the same here as we are doing for `"eta"` (e.g. have a "accepts_generator" flag and then set it if relevant?)

Reply:
The cleaner option that we should probably do somewhat soon is what @keturn proposed [here](https://github.com/huggingface/diffusers/issues/336#issue-1360766761), *i.e.* removing this scheduler specific logic and just renaming `**kwargs` to `**extra_step_kwargs`